src/controllers/cron/update_local_smartapi.js

```diff
@@ -153,24 +153,68 @@ const getOpsFromPredicatesEndpoints = async (specs) => {
 
 const updateSmartAPISpecs = async () => {
     const SMARTAPI_URL = 'https://smart-api.info/api/query?q=tags.name:translator&size=1000&sort=_seq_no&raw=1&fields=paths,servers,tags,components.x-bte*,info,_meta';
-    const res = await axios.get(SMARTAPI_URL, { headers: { 'User-Agent': userAgent } });
+    const overridesPath = path.resolve(__dirname, "../../config/smartapi_overrides.json");
     const localFilePath = path.resolve(__dirname, '../../../data/smartapi_specs.json');
-    const predicatesFilePath = path.resolve(__dirname, '../../../data/predicates.json');
-    const writeFunc = process.env.SYNC_AND_EXIT === "true" ? fs.writeFileSync : fs.writeFile;
-    if (process.env.API_OVERRIDE === "true") {
-        await getAPIOverrides(res.data);
+    let overrides;
+    try {
+        overrides = JSON.parse((await readFile(overridesPath)));
+    } catch (error) {
+        debug(`ERROR getting API Overrides file because ${error}`);
+        return;
     }
-    debug(`Retrieved ${res.data.total} SmartAPI records`);
-    //clean _score fields
-    const hits = res.data.hits;
-    hits.forEach(function(obj){ delete obj._score });
-    writeFunc(localFilePath, JSON.stringify({ hits: hits }), (err) => {
-        if (err) throw err;
-    });
-    const predicatesInfo = await getOpsFromPredicatesEndpoints(res.data.hits);
-    writeFunc(predicatesFilePath, JSON.stringify(predicatesInfo), (err) => {
-        if (err) throw err;
+    debug(overrides)
+    const res = await axios.get(SMARTAPI_URL, { headers: { 'User-Agent': userAgent } }).catch((err) => {
+        if(process.env.API_OVERRIDE === "true" && overrides.conf.only_overrides === true) {
+            debug(`SmartAPI request failed, creating specs from overrides config.`);
+            let hits = []
+            Object.keys(overrides.apis).map(async (id) => {
+                let override; 
+                try {
+                    const filepath = path.resolve(overrides.apis[id]);
+                    override = yaml.load(await readFile(filepath));
+                } catch (filerror) {
+                    debug(`ERROR getting local file override for API ID ${id} because ${filerror}`);
+                    return;
+                }
+                override._id = id;
+                override._meta = {
+                    date_created: undefined,
+                    last_updated: undefined,
+                    url: overrides.apis[id],
+                    username: undefined,
+                };
+                hits.push(override)
+                fs.writeFile(localFilePath, JSON.stringify({ hits: hits }), (err) => {
+                    if (err) throw err;
+                })
+            });
```

Comment:
You don't have to duplicate the functionality of `getAPIOverrides()` (which has more functionality that what you've added here). You can instead just create a mock `res.data` object containing an empty array `hits` and call `getAPIOverrides()` on it, letting the existing function handle override creation. Then you can pass those hits to the rest of the function.

This mocked `res.data` could be created prior to the request and overwritten with the actual `res.data` using your condition `if (await res) {}`.